Remove unnecessary MY_IP_RENEWAL_INTERVAL*

mfalkvidd opened the pull request:
Ethernet.maintain already checks when we need to renew, based on
the lease time given by the DHCP server.

https://www.arduino.cc/en/Reference/EthernetMaintain recommends
"just call it on every loop() invocation" so there is no need for
us to keep a separate timer.

mfalkvidd committed:
Remove unnecessary MY_IP_RENEWAL_INTERVAL*

Ethernet.maintain already checks when we need to renew, based on
the lease time given by the DHCP server.

https://www.arduino.cc/en/Reference/EthernetMaintain recommends
"just call it on every loop() invocation" so there is no
need for us to keep a separate timer.

mfalkvidd commented on core/MyGatewayTransportEthernet.cpp:
```diff
@@ -150,19 +150,10 @@ void gatewayTransportRenewIP(void)
 		GATEWAY_DEBUG(PSTR("!GWT:TRC:IP RENEW FAIL\n"));
-		return;
 	}
 	_w5100_spi_en(false);
```
Not sure why `_w5100_spi_en(false)` was only called in case of successful renewal before. IF it should be called at all, my guess is that it should be called regardless whether renewal was successful or failed

Yveaux replied:
gatewayTransportRenewIP is only called from gatewayTransportAvailable.
gatewayTransportAvailable starts with _w5100_spi_en(true);
With your change we return to gatewayTransportAvailable with SPE_EN set false.
This is likely to break the remaining code in gatewayTransportAvailable.

I would remove any call to _w5100_spi_en() from gatewayTransportRenewIP, so it never messes with the state of the SPI_EN line.

The SPI_EN handling in that file is horrible; I would almost advise to use gotos 😱

mfalkvidd replied:
To be clear: without my change, we return to gatewayTransportAvailable with SPI_EN set to false when the renewal was successful (which it is most of the time).

mfalkvidd committed:
Remove call to _w5100_spi_en

This function doesn't enable SPI, so it should not
disable it either. Let calling functions handle SPI
enable/disable.

mfalkvidd commented:
If we fail to renew repeatedly, we will spam the debug console with messages. Not sure if that will cause any problems.
